src/System.Net.HttpListener/tests/GetContextHelper.cs

```diff
@@ -60,7 +60,15 @@ public class Helpers
     {
         public static void WaitForSocketShutdown(Socket socket)
         {
-            while (SocketConnected(socket));
+            if (PlatformDetection.IsWindows || PlatformDetection.IsOSX)
+            {
+                socket.Shutdown(SocketShutdown.Both);
```

Comment:
We use the managed implementation on macOS as well, so that doesn't explain why the check is for IsWindows || IsOSX.